src/raps/actions/swap.ts

```diff
@@ -93,9 +93,13 @@ export const estimateSwapGasLimit = async ({
         WRAP_GAS_PADDING
       );
 
-      return gasLimit || String(quote?.defaultGasLimit) || String(default_estimate);
+      if (gasLimit === null || gasLimit === undefined || isNaN(Number(gasLimit))) {
+        return quote?.defaultGasLimit || default_estimate;
+      }
+
+      return gasLimit;
     } catch (e) {
-      return String(quote?.defaultGasLimit) || String(default_estimate);
+      return quote?.defaultGasLimit || default_estimate;
```

Comment:
my guess was this was causing an issue with the string case on quote?.defaultGasLimit when `undefined`